discovery-provider/src/queries/search.py

```diff
@@ -40,6 +43,145 @@ def search_full():
+                    updated_at desc
+            ) as t
+                where
+                tagstrip @@ to_tsquery(:query);
+            """
+        )
+        user_res = sqlalchemy.text(
+            f"""
+            select * from
+            (
+		select
+                    count(track_id),
+                    owner_id
+                from
+		(
+                    select
+                        strip(to_tsvector(tracks.tags)) as tagstrip,
+                        tags,
+                        track_id,
+                        owner_id
+                    from
+			tracks
+                    where
+                        (tags like :like_tags_query)
+			and (is_current is true)
+                    order by
+			updated_at desc
+                ) as t
+                where
+                        tagstrip @@ to_tsquery(:query)
+                group by
+                        owner_id
+                order by
+                        count desc
+            ) as usr
+            where
+                usr.count > :user_tag_count;
+            """
+        )
+    track_ids = session.execute(
+        track_res,
+        {
+            "query":search_str,
+            "like_tags_query":like_tags_str
+        }
+    ).fetchall()
+    user_ids = session.execute(
+        user_res,
+        {
+            "query":search_str,
+            "like_tags_query":like_tags_str,
+            "user_tag_count": user_tag_count
+        }
+    ).fetchall()
+
+    # track_ids is list of tuples - simplify to 1-D list
+    track_ids = [i[0] for i in track_ids]
+
+    # user_ids is list of tuples - simplify to 1-D list
+    user_ids = [i[1] for i in user_ids]
+
+    followee_count_dict = get_followee_count_dict(session, user_ids)
+
+    tracks = (
+        session.query(Track)
+        .filter(
+            Track.is_current is True,
+            Track.is_delete is False,
+            Track.track_id.in_(track_ids),
+        )
+        .all()
+    )
+    tracks = helpers.query_result_to_list(tracks)
+
+    track_play_counts = get_track_play_counts(track_ids)
+
+    users = (
+        session.query(User)
+        .filter(
+            User.is_current is True,
+            User.is_ready is True,
+            User.user_id.in_(user_ids)
+        )
+        .all()
+    )
+    users = helpers.query_result_to_list(users)
+    for user in users:
+        user_id = user["user_id"]
+        user[response_name_constants.follower_count] = followee_count_dict.get(user_id, 0)
+
+    followee_sorted_users = \
+        sorted(users, key=lambda i: i[response_name_constants.follower_count], reverse=True)
+
+    for track in tracks:
+        track_id = track["track_id"]
+        track[response_name_constants.play_count] = track_play_counts.get(track_id, 0)
+
+    play_count_sorted_tracks = \
+        sorted(tracks, key=lambda i: i[response_name_constants.play_count], reverse=True)
+
+    # Add pagination parameters to track results
+    play_count_sorted_tracks = \
```

Comment:
@hareeshnagaraj @piazz should followee_sorted_users also be paginated?

Reply:
done